xivo_bus/resources/common/abstract.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022-2022 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import re
+
+from inspect import isabstract
+from abc import ABCMeta, abstractmethod
+from six import with_metaclass
+
+
+def camel_to_snake(name):
+    name = name.split('Event')[0]
+    name = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', name).lower()
+
+
+class EventMeta(ABCMeta):
+    def __new__(mcs, name, bases, attrs):
+        cls = super(EventMeta, mcs).__new__(mcs, name, bases, attrs)
+        if not isabstract(cls) and 'name' not in attrs:
+            setattr(cls, 'name', camel_to_snake(name))
+        return cls
+
+
+class AbstractEvent(with_metaclass(EventMeta, object)):
+    __slots__ = ('content',)
+
+    def __init__(self, content=None):
+        self.content = content or {}
+
+    def __eq__(self, other):
+        return (
+            self.__class__ == other.__class__
+            and self.content == other.content
+            and vars(self) == vars(other)
+        )
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self):
```

Comment:
`__repr__` should allow to an equivalent object to be created. It's implementation could be something like
```
def __repr__(self):
    return "%s(%r)" % (self.__class__, self.content)
```

Reply:
This comment is mainly an opinion. I think it would be more useful that having the same representation as the `__str__` method